File: boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) 
    {
        sort(people.begin(), people.end());
        int s = 0, e = people.size() - 1;
        int count = 0;
        while (s < e)
        {
            if (people[s] + people[e] > limit)
                e--;
            else
            {
                s++;
                e--;
            }
            count++;
        }
        if (s == e)
        {
            count++;
        }
        return count;
    }
};

File: leet-code-solutions/boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        N = len(people)
        l, r = 0, N - 1
        count = 0

        while l <= r:
            if people[l] + people[r] <= limit:
                l += 1

            r -= 1
            count += 1

        return count
